### PR TITLE
Add "get metarange", "get range" API endpoints

### PR DESCRIPTION
**Pull request by arielshaqed**
Returns data (currently just the address of the object) by ID.

### CODE REVIEW

File: graveler/graveler.go [hunk 1]
```diff
@@ -321,6 +334,15 @@ type VersionController interface {
 	SetHooksHandler(handler HooksHandler)
 }
 
+// Plumbing includes commands for fiddling more directly with graveler implementation
+// internals.
+type Plumbing interface {
+	// GetMetarange returns information where metarangeID is stored.
+	GetMetaRange(ctx context.Context, repositoryID RepositoryID, metaRangeID MetaRangeID) (MetaRangeData, error)
+	// GetRange returns information where rangeID is stored.
```
Comment (nopcoder) [inline graveler/graveler.go:342]: Consider calling it xxxStats returning xxxInfo
ex: MetaRangeStat -> MetaRangeInfo

File: api/controller.go [hunk 2]
```diff
@@ -1265,6 +1265,56 @@ func writeSymlinkToS3(params metadata.CreateSymlinkParams, repo *catalog.Reposit
 	return err
 }
 
+func (c *Controller) MetadataGetMetarangeHandler() metadata.GetMetaRangeHandler {
+	return metadata.GetMetaRangeHandlerFunc(func(params metadata.GetMetaRangeParams, user *models.User) middleware.Responder {
+		deps, err := c.setupRequest(user, params.HTTPRequest, []permissions.Permission{
+			{
+				// TODO(oz): May want tighter permissions here.
```
Comment (nopcoder) [inline api/controller.go:1272]: Remember to remove the todo after verifying

File: api/controller.go [hunk 3]
```diff
@@ -1265,6 +1265,56 @@ func writeSymlinkToS3(params metadata.CreateSymlinkParams, repo *catalog.Reposit
...
+		ret.Location = metarange.Address
+		return ret
+	})
+}
+
+func (c *Controller) MetadataGetRangeHandler() metadata.GetRangeHandler {
+	return metadata.GetRangeHandlerFunc(func(params metadata.GetRangeParams, user *models.User) middleware.Responder {
+		deps, err := c.setupRequest(user, params.HTTPRequest, []permissions.Permission{
+			{
+				// TODO(oz): May want tighter permissions here.
```
Comment (nopcoder) [inline api/controller.go:1297]: Remember to remove the todo after verifying

File: swagger.yml [hunk 4]
```diff
@@ -2154,6 +2154,64 @@ paths:
...
+        - metadata
+      operationId: getMetaRange
+      summary: return URI to a meta-range file
+      # TODO(ariels): Actually redirect to S3.
+      responses:
+        200:
+          description: meta-range URI
+          headers:
+            Location:
+              type: string
```
Comment (nopcoder) [inline swagger.yml:2178]: Why did we prefer to return the response just as a header for this case?
All of our API usually return the response as JSON body.
We can have `location` (bool) parameter to control if we like to get the response just as a location header for a client that doesn't want to parse the response body.

**Review by nopcoder — APPROVED** [review]
Looks great, with minor comments/suggestions and a question down at the end before you commit.

File: swagger.yml (hunk 4 above)
Comment (arielshaqed) [reply swagger.yml:2178]: This is *almost* a redirect action.  In fact, if we go ahead with signed URLs then it _will_ be a redirect.  I prefer to abuse HTTP semantics here, at least a bit.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Keeping the Location header, changing the names from "...Data" to "...Info".

**Comment by arielshaqed** [comment]
+ @ozkatz to look at permissions for these actions, please.

File: api/controller.go (hunk 2 above)
Comment (arielshaqed) [reply api/controller.go]: After f2fing, changed to require _both_ list objects _and_ read repo: it gives a cross between the 2.  E.g. with (just) read permission on underlying storage it gives list objects (and without it gives size of branch), which is list-objects-like.  But it _also_ gives the storage namespace for the repo, which is read-repo-like.  So best to require both.

**Review by arielshaqed — COMMENTED** [review]
Thanks!
Pulling once the tests settle.

**Pull request merged**